File: src/015. Lattice paths/015.lua
path = {}
m = 0
n = 0
for m = 0, 20 do
  for n = m, 20 do
    print("path " .. m .. " " .. n)
    if m == 0 then
      path[m .. "_" .. n] = 1
    elseif m == n then
      path[m .. "_" .. n] = path[(m - 1) .. "_" .. n] * 2
    else
      path[m .. "_" .. n] = path[(m - 1) .. "_" .. n] + path[m .. "_" .. (n - 1)]
    end
  end
end

print(path[20 .. "_" .. 20])

File: src/015. Lattice paths/015.rb
path = Hash.new(0)
0.upto(20) do |m|
  m.upto(20) do |n|
    path[(m.to_s + ':' + n.to_s).to_sym] = case m
                                             when 0 then 1
                                             when n then path[((m - 1).to_s + ':' + n.to_s).to_sym] * 2
                                             else        path[((m - 1).to_s + ':' + n.to_s).to_sym] + path[(m.to_s + ':' + (n - 1).to_s).to_sym]
                                           end
  end
end

puts path[:"20:20"]